frontend/app/src/components/helper/CollectionHandler.vue

```diff
@@ -1,20 +1,33 @@
 <script setup lang="ts">
-import { type PropType } from 'vue';
 import { type Collection } from '@/types/collection';
 import { getCollectionData, setupEntryLimit } from '@/utils/collection';
+import { useFrontendSettingsStore } from '@/store/settings/frontend';
 
-const props = defineProps({
-  collection: {
-    required: true,
-    type: Object as PropType<Collection<any>>
-  }
-});
+const props = defineProps<{
+  collection: Collection<any>;
```

Comment:
For vue 3 there are some guides on how to make generics work with components, in the case of slots it would be nice to have type inference. Unfortunately, the same thing does not seem to work with 2.7